### PR TITLE
[DOCS] Reformat distance feature query

### PR DESCRIPTION
**Pull request by jrodewig**
Updates the `distance_feature` query to use the new query format.

This creates separate sections for the example requests, parameters, and notes.

This is part of #40977, an effort to standardize documentation for query types.

### CODE REVIEW

File: docs/reference/query-dsl/distance-feature-query.asciidoc [hunk 1]
```diff
@@ -175,6 +146,83 @@ GET items/_search
...
+
+The `distance` is the absolute difference between the `origin` value and a
+document's field value.
+
+[[distance-feature-skip-hits]]
+===== Skip non-competitive hits
+Unlike the <<query-dsl-function-score-query,`function_score`>> query or other
+ways to change <<query-filter-context, relevance scores>>, the
+`distance_feature` query efficiently skips non-competitive hits when the
+<<search-uri-request,`track_total_hits`>> parameter is  `true`.
```
Comment (mayya-sharipova) [inline docs/reference/query-dsl/distance-feature-query.asciidoc]: is NOT set to `true`

File: docs/reference/query-dsl/distance-feature-query.asciidoc [hunk 2]
```diff
@@ -4,81 +4,37 @@
...
-    and a geo-point for geo_point fields. Date math (for example `now-1h`) is
-    supported for a date origin.
-
-`pivot`::
-    Required parameter. Defines the distance from origin at which the computed
-    score will equal to a half of the `boost` parameter. Must be
-    a `number+date unit` ("1h", "10d",...) for date and date_nanos fields,
-    and a `number + geo unit` ("1km", "12m",...) for geo fields.
+Boosts the <<query-filter-context, relevance score>> of documents closer to a
+provided `origin` date or point.
```
Comment (mayya-sharipova) [inline docs/reference/query-dsl/distance-feature-query.asciidoc]: I think  it is important to include some variation of the below paragraph, as to my mind it nicely highlights the main usage for this query:

"For example, use this query if you want to give more weight to documents with dates closer to a certain date, or to documents with locations closer to a certain location."

**Review by mayya-sharipova — APPROVED** [review]
thanks @jrodewig

File: docs/reference/query-dsl/distance-feature-query.asciidoc (hunk 2 above)
Comment (jrodewig) [reply docs/reference/query-dsl/distance-feature-query.asciidoc]: Re-added with 232a365d405.

File: docs/reference/query-dsl/distance-feature-query.asciidoc (hunk 1 above)
Comment (jrodewig) [reply docs/reference/query-dsl/distance-feature-query.asciidoc]: Fixed with 20a04920cbc.

File: docs/reference/query-dsl/distance-feature-query.asciidoc [hunk 3]
```diff
@@ -146,17 +116,18 @@ GET items/_search
...
 // CONSOLE
-// TEST[continued]
 
-We can look for all chocolate items, but we also want chocolates
-that are produced locally (closer to our geo origin)
-come first in the result list.
+[[distance-feature-query-distance-ex]]
+====== Boost documents based on location
+The following `bool` search returns documents with a `name` value of
+`chocolate`.The search also uses the `distance_feature` query to increase the
```
[the remark this answers is not in the document]
Comment (jrodewig) [reply docs/reference/query-dsl/distance-feature-query.asciidoc]: Thanks for catching this. Fixed with 6ff193dff44.

**Pull request merged**